src/Compilers/Core/Portable/SourceFileResolver.cs

```diff
@@ -62,9 +62,9 @@ public SourceFileResolver(ImmutableArray<string> searchPaths, string baseDirecto
                     }
 
                     var value = kv.Value;
-                    if (value == null || value.Length == 0)
+                    if (value == null)
                     {
-                        throw new ArgumentException(CodeAnalysisResources.EmptyValueInPathMap, nameof(pathMap));
+                        throw new ArgumentNullException(nameof(PathMap));
```

Comment:
> Wouldn't this exception say that pathMap parameter in null? Wouldn't this be misleading?

I was following the style of the code 3 lines up from this.

```
  throw new ArgumentException(CodeAnalysisResources.EmptyKeyInPathMap, nameof(pathMap));
```

>BTW, why there is no error for nameof here, the first letter of the argument is capital?

There's a property that has the capitalized version.  I still fixed this though.